src/DeploymentTemplate.ts

```diff
@@ -323,6 +334,70 @@ export class DeploymentTemplate {
         return functionCounts;
     }
 
+    /**
+     * Gets info about schema usage, useful for telemetry
+     */
+    public getResourceUsage(): Histogram {
+        const resourceCounts = new Histogram();
+
+        // Collect all resources used
+        const resources: Json.ArrayValue | null = this._topLevelValue ? Json.asArrayValue(this._topLevelValue.getPropertyValue(templateKeys.resources)) : null;
+        if (resources) {
+            traverseResources(resources, undefined);
+        }
+
+        return resourceCounts;
+
+        function traverseResources(resourcesObject: Json.ArrayValue, parentKey: string | undefined): void {
+            for (let resource of resourcesObject.elements) {
+                const resourceObject = Json.asObjectValue(resource);
+                if (resourceObject) {
+                    const resourceType = Json.asStringValue(resourceObject.getPropertyValue(templateKeys.resourceType));
+                    const apiVersion = Json.asStringValue(resourceObject.getPropertyValue(templateKeys.resourceApiVersion));
+                    if (resourceType && apiVersion) {
+                        const isTypeAnExpression = resourceType.unquotedValue.trim().startsWith('[');
```

Comment:
technically there are more rules to being an expression.  Is this a shortcut just for type/apiVersion since those would never otherwise have `[`?  Or should we be more precise about checking for an expression?

Reply:
Yeah, this is just a shortcut, it's unlikely the other rules would be used here.